### ISSUE TITLE
Date QueryParam serialization

### ISSUE THREAD

**Issue by kvpt** [tmenier/Flurl#582]
I found another different behavior between Flurl 2 and 3, I didn't find it in the breaking changes list so I think it'a regression.
When passing a DateTime as QueryParam the value are not url encoded like in Flurl 2.

```cs
using (HttpTest httpTest = new HttpTest())
{
	await "https://api.com"
	    .SetQueryParam("date", DateTime.Today)
	    .GetAsync();

	httpTest.CallLog.Last().Request.Url.Query.Dump();
}
```
With Flurl 2
> date=2020-12-05T00%3A00%3A00.0000000%2B01%3A00

With Flurl 3
> date=2020-12-05T00:00:00.0000000+01:00

If I pass the date as ISO string the date are correctly encoded like in Flurl 2.

**Comment by tmenier** [comment]
This appears to be legit, but should be an easy fix. Thanks for reporting, I'll get it out in the next maintenance release.

**Comment by tmenier** [comment]
Fixed & released in 3.0.1
